contracts/farm/LiquidityMiningV1.sol

```diff
@@ -705,4 +705,73 @@ contract LiquidityMiningV1 is ILiquidityMining, LiquidityMiningStorageV1 {
 		UserInfo memory ui = getUserInfo(_poolToken, _user);
 		return ui.amount;
 	}
+
+	/**
+	 * @notice returns arrays with all the pools on the contract
+	 */
+	function getPoolInfoListArray()
+		external
+		view
+		returns (
+			address[] memory _poolToken,
+			uint96[] memory _allocationPoints,
+			uint256[] memory _lastRewardBlock
+		)
+	{
+		uint256 length = poolInfoList.length;
+		for (uint256 i = 0; i < length; i++) {
+			_poolToken[i] = address(poolInfoList[i].poolToken);
+			_allocationPoints[i] = poolInfoList[i].allocationPoint;
+			_lastRewardBlock[i] = poolInfoList[i].lastRewardBlock;
+		}
+		return (_poolToken, _allocationPoints, _lastRewardBlock);
+	}
+
+	/**
+	 * @notice returns all pools that a user has on the contract, the poolId it's the index of arrays
+	 * @param _user the address of the user
+	 */
+	function getUserInfoListArray(address _user)
+		external
+		view
+		returns (
+			uint256[] memory _amount,
+			uint256[] memory _rewardDebt,
+			uint256[] memory _accumulatedReward
+		)
+	{
+		uint256 length = poolInfoList.length;
+		for (uint256 i = 0; i < length; i++) {
+			_amount[i] = userInfoMap[i][_user].amount;
+			_rewardDebt[i] = userInfoMap[i][_user].rewardDebt;
+			_accumulatedReward[i] = userInfoMap[i][_user].accumulatedReward;
+		}
+		return (_amount, _rewardDebt, _accumulatedReward);
+	}
+
+	/**
+	 * @notice reset user info from a pool, this function is called by LiquidityMiningV2 after migration
+	 * @param _user the addres of the user
+	 * @param _poolId the poolId to be reset
+	 */
+	function resetUser(address _user, uint256 _poolId) external onlyAuthorized {
+		UserInfo storage userInfo = userInfoMap[_poolId][_user];
+		userInfo.amount = 0;
+		userInfo.rewardDebt = 0;
+		userInfo.accumulatedReward = 0;
+	}
+
+	/**
+	 * @notice send all funds from this contract to LiquidityMiningV2
+	 */
+	function migrateFunds(address _lmContract) external onlyAuthorized {
```

Comment:
I would add here that the `gracePeriod` should have finished before allowing this operation to execute

Reply:
Yes, I added that task to the board because we don't have that modifier yet

Reply:
Ah, but we have the state variable already